fretboard/src/main/java/mozilla/components/service/fretboard/Fretboard.kt

```diff
@@ -14,21 +14,24 @@ class Fretboard(
     private val source: ExperimentSource,
     private val storage: ExperimentStorage
 ) {
-    internal var experiments: List<Experiment> = listOf()
-        private set
+    private var experiments: List<Experiment> = listOf()
+    private var experimentsLoaded: Boolean = false
 
     /**
      * Loads experiments from local storage
      */
     fun loadExperiments() {
         experiments = storage.retrieve()
+        experimentsLoaded = true
     }
```

Comment:
You may want to add @Synchronized to both methods now. load() and update() may be called on different threads and without synchronization they may a cached value:
http://blog.tremblay.pro/2017/08/data-visibility.html